cmd/daily_matrix/daily_matrix.go

```diff
@@ -102,22 +112,67 @@ func saveGeneratedHtml(generated_html []byte, f *Flags) error {
 }
 
 func populateTestFromFinished(test *v1.TestResult, test_finished artifacts.ArtifactResult) error {
-	test.Passed = test_finished.Json["passed"].(bool)
-	test.Result = test_finished.Json["result"].(string)
-	ts := test_finished.Json["timestamp"].(float64)
-	test.FinishDate = time.Unix(int64(ts), 0).Format("2006-01-02 15:04")
-
+	if test_finished.Json["passed"] != nil {
+		test.Passed = test_finished.Json["passed"].(bool)
+	} else {
+		test.Passed = false
+	}
+	if test_finished.Json["result"] != nil {
+		test.Result = test_finished.Json["result"].(string)
+	} else {
+		test.Result = "N/A"
+	}
+	if test_finished.Json["timestamp"] != nil {
+		ts := test_finished.Json["timestamp"].(float64)
+		test.FinishDate = time.Unix(int64(ts), 0).Format("2006-01-02 15:04")
+	} else {
+		test.FinishDate = "N/A"
+	}
 	return nil
 }
 
 func populateTestFromStepFinished(test *v1.TestResult, step_test_finished artifacts.ArtifactResult) error {
 	test.StepExecuted = true
-	test.StepPassed = step_test_finished.Json["passed"].(bool)
-	test.StepResult = step_test_finished.Json["result"].(string)
+	if step_test_finished.Json["passed"] != nil {
+		test.StepPassed = step_test_finished.Json["passed"].(bool)
+	} else {
+		test.StepPassed = false
+	}
+	if step_test_finished.Json["result"] != nil {
+		test.StepResult = step_test_finished.Json["result"].(string)
+	} else {
+		test.StepResult = "N/A"
+	}
+
+	return nil
+}
+
+func populateTestFromToolboxLogs(test *v1.TestResult, toolbox_logs map[string]artifacts.JsonArray) error {
+	test.Ok = 0
+	test.Failures = 0
+	test.Ignored = 0
+
+	for toolbox_step_name, toolbox_step_json := range toolbox_logs {
+
+		stats := toolbox_step_json[len(toolbox_step_json)-1].(map[string]interface{})["stats"].(map[string]interface{})["localhost"].(map[string]interface{})
+		ok := int(stats["ok"].(float64))
+		failures := int(stats["failures"].(float64))
+		ignored := int(stats["ignored"].(float64))
+		log.Debugf("Step %s: ok %d, failures %d, ignored %d", toolbox_step_name, ok, failures, ignored)
+
+		test.ToolboxStepsResults = append(test.ToolboxStepsResults,
+			v1.ToolboxStepResult{toolbox_step_name, ok, failures, ignored}, )
+
+		test.Ok = test.Ok + ok
```

Comment:
Would this be better written as:
```
test.Ok += ok
```
?  Similarly below.

Reply:
hum, I don't know how I fooled myself thinking it didn't work in Go :#
fixed 
(I'm still frustrated that there is no ternary operator `<test>? <true> : <false>` or `<true> if <test> else <false>` available :#)

Reply:
> (I'm still frustrated that there is no ternary operator `<test>? <true> : <false>` or `<true> if <test> else <false>` available :#)

Exactly!  Nice to see I'm not the only one missing it in golang. :)